File: backend/app/schemas/intel.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional


class IntelInitRequest(BaseModel):
    name: str = "军事情报分析演示"
    description: Optional[str] = None


class IntelInitResponse(BaseModel):
    ontology_id: str
    name: str


class IntelSubmitRequest(BaseModel):
    intel_text: str
    label: Optional[str] = None  # 不填则自动 T1/T2/T3


class IntelSubmitResponse(BaseModel):
    snapshot_id: str
    task_id: str
    status: str  # "extracting"


class IntelSnapshotOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    label: str
    intel_text: str
    entity_count: int
    relation_count: int
    danger_score: float
    danger_level: str
    recommendations: list[str]
    status: str
    created_at: datetime


class GraphData(BaseModel):
    nodes: list[dict]
    edges: list[dict]
    neo4j_available: bool = False
    fallback: Optional[str] = None


class IntelAssessResponse(BaseModel):
    ontology_id: str
    ontology_name: str
    danger_level: str
    danger_score: float
    recommendations: list[str]
    entity_count: int
    relation_count: int
    snapshots: list[IntelSnapshotOut]
    graph: GraphData


class IntelForwardRequest(BaseModel):
    """手动转发请求 — 传入情报文本，分析后推送到外部 agent"""
    intel_text: str


class IntelForwardResponse(BaseModel):
    success: bool
    message: str
    webhook_url: str
    webhook_task_id: str | None = None  # Celery 异步任务 ID，用于追踪推送状态
    payload: dict  # 实际发送给外部 agent 的内容

File: frontend/src/types/intel.ts
export interface IntelSnapshot {
  id: string
  label: string
  intel_text: string
  entity_count: number
  relation_count: number
  danger_score: number
  danger_level: 'low' | 'medium' | 'high' | 'critical'
  recommendations: string[]
  status: 'extracting' | 'completed' | 'failed'
  created_at: string
}

export interface GraphData {
  nodes: Array<{
    id: string
    labels: string[]
    properties: Record<string, unknown>
  }>
  edges: Array<{
    id: string
    source: string
    target: string
    type: string
  }>
  neo4j_available: boolean
  fallback?: string
}

export interface IntelAssessData {
  ontology_id: string
  ontology_name: string
  danger_level: string
  danger_score: number
  recommendations: string[]
  entity_count: number
  relation_count: number
  snapshots: IntelSnapshot[]
  graph: GraphData
}

export interface IntelInitResponse {
  ontology_id: string
  name: string
}

export interface IntelSubmitResponse {
  snapshot_id: string
  task_id: string
  status: string
}

export const DANGER_LABELS: Record<string, string> = {
  low: '低威胁',
  medium: '中等威胁',
  high: '高威胁',
  critical: '严重威胁',
}

export const DANGER_COLORS: Record<string, string> = {
  low: '#10b981',
  medium: '#f59e0b',
  high: '#f97316',
  critical: '#ef4444',
}
